### PR TITLE
Refactor date/timestamp formatting utilities

### PR DESCRIPTION
**Pull request by robertknight**
This is some general refactoring/cleanup of the date/time formatting utilities in the client that I did while looking into modules with low test coverage (`date.js` had no tests). This isn't a dependency of other work so can be reviewed whenever.

The main change is to combine `sidebar/util/date.js` and `sidebar/util/time.js`. It was confusing to have two separate modules when there wasn't a clear separation of "time utilities" vs "date utilities". This also eliminates some redundant caching logic. The names of the existing functions were renamed so that they are cohesive in the new module.

The one functional change is to remove code paths that only existed for IE 11 and Safari <= 9.0, which we don't support any more.

 - Combine `date.js` and `time.js`
 
 - Rename `toFuzzyString` and `format` to `formatRelativeDate` and
   `formatDate` respectively to make the differnce between them more
   obvious

 - Add missing test for `formatDate`

 - Remove the fallback code paths for old browsers that don't support
   `Intl.DateTimeFormat`. All of our currently supported browsers do.

 - Remove unused `date` parameter from `decayingInterval` callback. The
   only consumer in `AnnotationTimestamps` didn't use it.

 - Improve types in `time.js`

### CODE REVIEW

File: src/sidebar/util/time.js [hunk 1]
```diff
@@ -224,23 +217,46 @@ export function decayingInterval(date, callback) {
...
+export function formatDate(date) {
+  return format(date, {
+    year: 'numeric',
+    month: 'short',
+    day: '2-digit',
+    weekday: 'long',
+    hour: '2-digit',
+    minute: '2-digit',
+  });
 }
```
Comment (esanzgar) [inline src/sidebar/util/time.js:262]: I suggest to add a `Int` test seam here as well.

```suggestion
 * @param {Intl} [Intl] - Test seam. JS `Intl` API implementation.
 * @return {string}
 */
export function formatDate(date, Intl) {
  return format(
    date,
    {
      year: 'numeric',
      month: 'short',
      day: '2-digit',
      weekday: 'long',
      hour: '2-digit',
      minute: '2-digit',
    },
    Intl
  );
}
```

**Review by esanzgar — APPROVED** [review]
I added a suggestion for the testing the `formatDate`.

For `formatRelativeDate`, we should consider using `Intl.RelativeTimeFormat` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat) to support internationalitation:

```
new Intl.RelativeTimeFormat("de-DE", {
    localeMatcher: "best fit", // other values: "lookup"
    numeric: "auto", // other values: "auto"
    style: "long", // other values: "short" or "narrow"
}).format(-3, 'day')

"vor 3 Tagen"

new Intl.RelativeTimeFormat("de-DE", {
    localeMatcher: "best fit", // other values: "lookup"
    numeric: "auto", // other values: "auto"
    style: "long", // other values: "short" or "narrow"
}).format(-1, 'day')

"gestern"
```

**Comment by robertknight** [comment]
> For formatRelativeDate, we should consider using Intl.RelativeTimeFormat (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat) to support internationalitation:

Indeed. Out of scope for this PR, but definitely worth considering in future. Safari support is very recent (iOS 14), but it is supported at least. We could potentially use non-relative dates as a fallback for browsers that don't support it.

File: src/sidebar/util/test/time-test.js [hunk 2]
```diff
@@ -220,11 +202,25 @@ describe('sidebar.util.time', function () {
...
+      const formatted = formatDate(date);
+
+      // The exact format will depend on the system locale and browser, but it
+      // should contain at least the following tokens.
+      const tokens = ['2020', '04', '03', '02'];
+      tokens.forEach(token =>
+        assert.match(formatted, new RegExp(`\\b${token}\\b`))
+      );
+    });
+  });
```
[the remark this answers is not in the document]
Comment (robertknight) [reply src/sidebar/util/test/time-test.js]: I have applied this suggestion. There is some risk here that the exact output changes in a minor way in a future browser release that breaks the tests, but it should be easy to adjust the tests if that happens.

**Pull request merged**